webapi/Plugins/Chat/ApiConnectorPlugin.cs

```diff
@@ -0,0 +1,193 @@
+using System;
+using System.ComponentModel;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel;
+using System.Collections.Generic;
+using System.Net.Http;
+using Microsoft.Extensions.Configuration;
+
+namespace CopilotChat.WebApi.Plugins.APIConnector;
+
+/// <summary>
+/// This class is a plugin that connects to an API.
+/// </summary>
+public sealed class ApiConnectorPlugin
+{
+    private readonly string _bearerToken;
+
+    private readonly ILogger _logger;
+
+    private readonly IHttpClientFactory _clientFactory;
+
+    private readonly string _clientId;
+
+    private readonly string _clientSecret;
+
+    private readonly string _tenantId;
+
+    private readonly string _authority;
+
+
+
+    //
+    // Summary:
+    //     Initializes a new instance of the Microsoft.SemanticKernel.Plugins.TaskListPlugin
+    //     class.
+    //
+    // Parameters:
+    //   bearerToken:
+    //     The bearer token to use for the API call.
+    //
+    //   clientFactory:
+    //     The factory to use to create HttpClient instances.
+    //
+    //   loggerFactory:
+    //     The factory to use to create ILogger instances.
+    public ApiConnectorPlugin(string bearerToken, IHttpClientFactory clientFactory, ILoggerFactory? loggerFactory = null)
+    {
+        this._bearerToken = bearerToken ?? throw new ArgumentNullException(bearerToken);
+
+        this._clientFactory = clientFactory;
+
+        ILogger logger;
+        if (loggerFactory == null)
+        {
+            ILogger instance = NullLogger.Instance;
+            logger = instance;
+        }
+        else
+        {
+            logger = loggerFactory.CreateLogger(typeof(ApiConnectorPlugin));
+        }
+
+        this._logger = logger;
+
+        //read configuration from appsettings file
```

Comment:
You should be able to use dependency injection to pass down these settings from the ChatController, rather than initializing a new Configuration

Reply:
fixed - taking logger from chat controller